### PR TITLE
Implement _create with unit tests

### PR DESCRIPTION
**Pull request by Padraic-O-Mhuiris**
- **`_create` implementation**: Validates attributes (count, sort order, canonical encoding, string size), expiry, mints a deterministic entity key from a monotonic per-owner nonce, computes the two-level EIP-712 hash (coreHash + entityHash), and stores a 3-slot on-chain Commitment. Full entity data (payload, contentType, attributes) remains in calldata — not emitted in the event.
- **Existence check uses `creator != address(0)`** instead of `createdAt != 0` — `msg.sender` is guaranteed non-zero, no block number assumptions needed.
- **Canonical attribute encoding enforced**: STRING attributes must have `fixedValue == 0`, UINT/ENTITY_KEY must have empty `stringValue`. Prevents multiple encodings producing different hashes for semantically identical attributes.
- **`currentBlock()` threaded as parameter**: `execute()` calls `currentBlock()` once and passes it to all op functions, avoiding redundant derivation.
- **Removed `EntityAlreadyExists` check**: nonce monotonicity guarantees key uniqueness — the SLOAD was unreachable dead code.
- **Slimmed `EntityCreated` event**: removed `payload`, `contentType`, and `attributes` from the event. Off-chain indexers reconstruct from transaction calldata.
- **Removed `MAX_PAYLOAD_SIZE` constant**: payload is uncapped — gas is self-limiting and caller-paid. Kept `MAX_ATTRIBUTES` (32) and `MAX_STRING_ATTR_SIZE` (1KB) as application-level bounds, moved to `EntityHashing` library.
- **25 unit tests** for `_create` via harness (`test/unit/ops/Create.t.sol`): validation reverts, state mutations, key determinism, event emission, hash correctness, edge cases.

### CODE REVIEW

File: src/EntityRegistry.sol [hunk 1]
```diff
@@ -173,40 +194,135 @@ contract EntityRegistry is EIP712("Arkiv EntityRegistry", "1") {
...
+    ///                expiresAt)
+    ///
+    /// Storage: writes a Commitment struct (3 slots) keyed by entityKey.
+    /// Key uniqueness is guaranteed by the monotonic per-owner nonce —
+    /// no existence check is needed.
+    function _create(EntityHashing.Op calldata op, BlockNumber current)
+        internal
+        returns (bytes32 key, bytes32 entityHash_)
+    {
+        // TODO: contentType validation (e.g. non-empty, allowlist of MIME types).
```
Comment (matthiaszimmermann) [inline src/EntityRegistry.sol]: the only restriction for mime type should be length and char set, i think there is an official spec for mime type definition. whatever we do here should likely also be informed by that one

File: src/EntityRegistry.sol [hunk 2]
```diff
@@ -173,40 +194,135 @@ contract EntityRegistry is EIP712("Arkiv EntityRegistry", "1") {
...
+        for (uint256 i = 0; i < op.attributes.length; i++) {
+            bytes32 name = ShortString.unwrap(op.attributes[i].name);
+
+            if (name == bytes32(0)) {
+                revert EntityHashing.EmptyAttributeName(i);
+            }
+
+            if (i > 0) {
+                bytes32 prev = ShortString.unwrap(op.attributes[i - 1].name);
+                if (name <= prev) {
```
Comment (matthiaszimmermann) [inline src/EntityRegistry.sol]: should likely be name < prev i guess. key names must be unique per entity.

File: src/EntityRegistry.sol [hunk 3]
```diff
@@ -173,40 +194,135 @@ contract EntityRegistry is EIP712("Arkiv EntityRegistry", "1") {
...
+            }
+        }
+
+        // expiresAt must be strictly after the current block. Equality is
+        // rejected because the entity would already be expirable in this block.
+        if (op.expiresAt <= current) {
+            revert EntityHashing.ExpiryInPast(op.expiresAt, current);
+        }
+
+        // Derive a globally unique entity key: keccak256(chainId, registry, owner, nonce).
```
Comment (matthiaszimmermann) [inline src/EntityRegistry.sol]: prefer to have an explicit (internal) key generation function that is used here. allows for better readability and documentation of the key generation than buried inside the creation function. also allows to add tests that only target key generation.

File: src/EntityRegistry.sol [hunk 4]
```diff
@@ -173,40 +194,135 @@ contract EntityRegistry is EIP712("Arkiv EntityRegistry", "1") {
...
+        _commitments[key] = EntityHashing.Commitment({
+            creator: msg.sender,
+            createdAt: current,
+            updatedAt: current,
+            expiresAt: op.expiresAt,
+            owner: msg.sender,
+            coreHash: coreHash_
+        });
+
+        emit EntityCreated(key, msg.sender, entityHash_, op.expiresAt);
```
Comment (matthiaszimmermann) [inline src/EntityRegistry.sol]: swap order of last two arguments. 1st three args are indexed and searchable (see more use cases that want to search for expiresAt than use cases that search for entity hash

File: src/EntityRegistry.sol (hunk 1 above)
Comment (Padraic-O-Mhuiris) [reply src/EntityRegistry.sol]: I've extended this TODO comment where there can be bitmap single pass solution. Best do it in another PR

File: src/EntityRegistry.sol (hunk 2 above)
Comment (Padraic-O-Mhuiris) [reply src/EntityRegistry.sol]: This validation has moved into the `coreHash` and `attributeHash` function

**Comment by matthiaszimmermann** [comment]
good to merge
